src/opt/Optimizer/Optimizer.hs

```diff
@@ -128,3 +129,28 @@ dropBorrowBlocks = extend dropBorrowBlock
            ,decls = [([VarNoType name], target)]
            ,body}
       dropBorrowBlock e = e
+
+forwardExistingFuture = extend forwardExistingFuture'
+  where
+    forwardExistingFuture' e@(Forward{emeta, forwardExpr=arg@VarAccess{}})=Forward{emeta=emeta', forwardExpr=newExpr}
```

Comment:
I don't think this is general enough. It should capture every case that is not a method call and not a future chaining. So those cases should be matched first and not transformed, then any other case should be handled by this rule. All those cases will be handled the same.
```
forward(e) ==> forward(e ~~> id)
```